### PR TITLE
 implemented xi function

### PR DESCRIPTION
**Pull request by abh2k**
* functions
  * Implemented Riemann Xi function

### CODE REVIEW

**Comment by abhinav28071999** [comment]
@smichr @oscarbenjamin I don't know what the error is, please help me to resolve this.

File: sympy/functions/special/zeta_functions.py [hunk 1]
```diff
@@ -582,6 +582,40 @@ def _eval_rewrite_as_zeta(self, s, **kwargs):
...
+
+
+    References
+    ==========
+
+    .. [1] https://en.wikipedia.org/wiki/Riemann_Xi_function
+
+    """
+
+    from sympy import gamma
```
Comment (sylee957) [inline sympy/functions/special/zeta_functions.py]: Probably you should import gamma inside each functions, if there is an issue with the cyclic import.
But why does putting the import on top of the file does not work?

File: sympy/functions/special/zeta_functions.py [hunk 2]
```diff
@@ -582,6 +582,40 @@ def _eval_rewrite_as_zeta(self, s, **kwargs):
...
+    @classmethod
+    def eval(cls, s):
+        z = zeta(s)
+        if not z.has(zeta):
```
Comment (sylee957) [inline sympy/functions/special/zeta_functions.py]: Why does `z.has(zeta)` decides the evaluation?

Comment (jksuom) [reply sympy/functions/special/zeta_functions.py]: Could this be `if not isinstance(z, zeta):`?

File: sympy/functions/special/zeta_functions.py [hunk 3]
```diff
@@ -582,6 +582,40 @@ def _eval_rewrite_as_zeta(self, s, **kwargs):
...
+            return s * (s - 1) * gamma(s/2) * z / (2 * pi**(s / 2))
```
Comment (jksuom) [inline sympy/functions/special/zeta_functions.py]: The number of spaces could be reduced (though there are many possibilities).

File: sympy/functions/special/zeta_functions.py [hunk 4]
```diff
@@ -582,6 +582,40 @@ def _eval_rewrite_as_zeta(self, s, **kwargs):
...
 
+class riemann_xi(Function):
+    r"""
+    Riemann Xi function.
+
+    Examples
+    ========
+
+    The Riemann Xi function is closely related to the Riemann zeta function.
+    The zeros of Riemann Xi function is precisely the non-trivial zeros
```
Comment (czgdp1807) [inline sympy/functions/special/zeta_functions.py]: `is` -> `are`?

File: sympy/functions/special/zeta_functions.py [hunk 5]
```diff
@@ -582,6 +582,40 @@ def _eval_rewrite_as_zeta(self, s, **kwargs):
...
+    of the zeta function.
+
+    >>> from sympy import riemann_xi, zeta
+    >>> from sympy.abc import s
+    >>> riemann_xi(s).rewrite(zeta)
+    pi**(-s/2)*s*(s - 1)*gamma(s/2)*zeta(s)/2
+
+
```
Comment (czgdp1807) [inline sympy/functions/special/zeta_functions.py]: Remove.

**Pull request merged**